Source/WebCore/dom/ViewTransition.cpp

```diff
@@ -859,12 +860,26 @@ RenderViewTransitionCapture* ViewTransition::viewTransitionNewPseudoForCapturedE
     return nullptr;
 }
 
+// https://drafts.csswg.org/css-view-transitions/#page-visibility-change-steps
+void ViewTransition::visibilityStateChanged()
+{
+    if (!document())
+        return;
+
+    if (protectedDocument()->hidden() && protectedDocument()->activeViewTransition() == this) {
+        protectedDocument()->clearInboundViewTransitionParams();
```

Comment:
Inbound view transition parameters get moved into the ViewTransition when it gets created, so they'll always be usually be nullptr already here (unless this is a same-doc VT, created before the `pagereveal` event).

I think the Document just needs to clear the params in the state change, or make the params themselves a `VisibilityChangeClient`.

Reply:
I can fix this in a follow-up, since it's level 2 specific.